clippy_lints/src/option_env_unwrap.rs

```diff
@@ -0,0 +1,56 @@
+use crate::utils::{is_direct_expn_of, span_lint_and_help};
+use if_chain::if_chain;
+use rustc::lint::in_external_macro;
+use rustc_lint::{EarlyContext, EarlyLintPass};
+use rustc_session::{declare_lint_pass, declare_tool_lint};
+use syntax::ast::*;
+
+declare_clippy_lint! {
+    /// **What it does:** Checks for usage of `option_env!(...).unwrap()` and
+    /// suggests usage of the `env!` macro.
+    ///
+    /// **Why is this bad?** Unwrapping the result of `option_env!` will panic
+    /// at run-time if the environment variable doesn't exist, whereas `env!`
+    /// catches it at compile-time.
+    ///
+    /// **Known problems:** None.
+    ///
+    /// **Example:**
+    ///
+    /// ```rust,no_run
+    /// let _ = option_env!("HOME").unwrap();
+    /// ```
+    ///
+    /// Is better expressed as:
+    ///
+    /// ```rust,no_run
+    /// let _ = env!("HOME");
+    /// ```
+    pub OPTION_ENV_UNWRAP,
+    correctness,
+    "using `option_env!(...).unwrap()` to get environment variable"
+}
+
+declare_lint_pass!(OptionEnvUnwrap => [OPTION_ENV_UNWRAP]);
+
+impl EarlyLintPass for OptionEnvUnwrap {
+    fn check_expr(&mut self, cx: &EarlyContext<'_>, expr: &Expr) {
+        if_chain! {
+            if !in_external_macro(cx.sess, expr.span);
+            if let ExprKind::MethodCall(path_segment, args) = &expr.kind;
+            let method_name = path_segment.ident.as_str();
+            if method_name == "expect" || method_name == "unwrap";
+            if let ExprKind::Call(caller, _) = &args[0].kind;
+            if is_direct_expn_of(caller.span, "option_env").is_some();
```

Comment:
@lzutao @flip1995 Hey is there a way to know the path of a macro? I want to make sure `option_env` is not a user-defined macro but I couldn't find anyway to get its path.

Reply:
Not that I'm aware of. At least not in an Early/LatePassLint. It would be possible with a `PreExpansionPass` lint, but we can't  use this kind here. It shouldn't matter to much though. Redefining compiler built-in macros is questionable anyway.

Reply:
Sounds good. Thank you.